internal/plugintest/helper.go

```diff
@@ -124,6 +126,50 @@ func (h *Helper) NewWorkingDir(ctx context.Context) (*WorkingDir, error) {
 		return nil, fmt.Errorf("unable to create terraform-exec instance: %w", err)
 	}
 
+	err = tf.SetDisablePluginTLS(true)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to disable terraform-exec plugin TLS: %w", err)
+	}
+
+	err = tf.SetSkipProviderVerify(true) // Only required for Terraform CLI 0.12.x
+
+	var mismatch *tfexec.ErrVersionMismatch
+	if err != nil && !errors.As(err, &mismatch) {
+		return nil, fmt.Errorf("unable to disable terraform-exec provider verification: %w", err)
+	}
+
+	if logPath := os.Getenv(EnvTfAccLogPath); logPath != "" {
+		logging.HelperResourceTrace(
+			ctx,
+			fmt.Sprintf("Setting terraform-exec log path via %s environment variable", EnvTfAccLogPath),
+			map[string]interface{}{logging.KeyTestTerraformLogPath: logPath},
+		)
+
+		if err := tf.SetLogPath(logPath); err != nil {
+			return nil, fmt.Errorf("unable to set terraform-exec log path (%s): %w", logPath, err)
+		}
+	}
+
+	// Similar to helper/logging.LogOutput() and
+	// terraform-plugin-log/tfsdklog.RegisterTestSink(), the TF_LOG_PATH_MASK
+	// environment variable should take precedence over TF_ACC_LOG_PATH.
+	if logPathMask := os.Getenv(EnvTfLogPathMask); logPathMask != "" {
+		// Escape special characters which may appear if we have subtests
+		testName := strings.Replace(t.Name(), "/", "__", -1)
+		logPath := fmt.Sprintf(logPathMask, testName)
+
+		logging.HelperResourceTrace(
+			ctx,
+			fmt.Sprintf("Setting terraform-exec log path via %s environment variable", EnvTfLogPathMask),
+			map[string]interface{}{logging.KeyTestTerraformLogPath: logPath},
+		)
+
+		if err := tf.SetLogPath(logPath); err != nil {
+			return nil, fmt.Errorf("unable to set terraform-exec log path (%s): %w", logPath, err)
+		}
+	}
```

Comment:
I think technically this is spot on, but there is an issue I can imagine could confuse users: if one sets both `TF_ACC_LOG_PATH` and `TF_LOG_PATH_MASK`, the latter wins (agreed) but they will see 2 log traces. This could lead to confusion as we know that the latter log is the one to trust, but a practitioner won't necessarily follow the logic.

Could we instead check for those env vars presence, before we commit with a log and the remaining logic that sets `logPath`?

Reply:
Updated the implementation -- does this now do what you had in mind?

Reply:
👨‍🍳 💋